### ISSUE TITLE
About influence of pretrained weights 

### ISSUE THREAD

**Issue by stillwaterman** [qubvel/segmentation_models#25]
hello qubvel, Is the classification model weights trained on Imagenet helpful to the effect of image segmentation? Have you done any related experiments?  If so, could you please show some experimental results? At last, do you think the weights of pre-trained backbone are helpful to the segmentation of remote sensing images? I am looking forward to your reply.

**Comment by qubvel** [comment]
Usefulness of pretrained weights for image segmentation (even for data from completely different domains) have been proven by number of competitions, you can find some of them on https://kaggle.com (the most popular data science completions platform). 
If you look at solutions of top places for such competitions as "SpaceNet" or "Urban 3D mapper" you will see that they use transfer learning.
To know mode about transfer learning for remote sensing data please read this https://arxiv.org/pdf/1801.05746.pdf.

**Comment by PhySci** [comment]
Hi everyone,
  @qubvel Many thanks for the repo - you have done a great work and it helps me a lot.
But I'm in doubt that recomended "useful trick" works well. In my experiments I saw decreasing of loss for freezed model. After unfreezeing loss became suddenly much higher. It looks like "set_trainable(model)" reinitializes everything, because it recompile the model from scratch. 

Please take a look at loss graph for example. Pink line shows loss of freezed model, green one is loss of unfreezed model
![loss](https://user-images.githubusercontent.com/10773733/47348287-cfe53900-d6b9-11e8-9fe5-3cb8b07668af.png)

**Comment by stillwaterman** [comment]
@qubvel  Thanks for  your reply! I have read the paper, the results show usefulness of pretrained weights, thanks. Then I have another question.  The pretrained weights are trained on 8 bits RGB images, if I use 16 bits RGB images to finetune, I want to know if the pretrained weights are still useful.  In my opinion,  I think Bacth Normalization can help us to normalize data distribution, so we can use  16 bits RGB images to finetune. I really want to know what your opinion is？

**Comment by qubvel** [comment]
@stillwaterman I think you are right and batchnorm can actually help, however, in my experiments pre-processing 16bit images to 8bit works better despite to some information loss. The best way to know is to make a set of experiments with your data :smile:

**Comment by qubvel** [comment]
@PhySci Keras has slightly strange behavior of batch normalization layer, it counts statistics even if freezed during training (however didn't use it during validation). So if you have data domain significantly differ from ImageNet data domain I recommend you unfreeze all batchnorm layers before training.

**Comment by stillwaterman** [comment]
@qubvel Thank you bro
